### DISCUSSION TITLE
https (Windows)

### DISCUSSION THREAD

**Discussion by ntenoutasse** [datalust/seq-tickets#1474] [Help] [answered]
Hi, we are moving seq from http to https (on Windows) but we would like to move the UI completely on https but keep the ingestion on http & https for some time and then when all the applications are deployed in prod to use the https only, remove the http entry point.
But as far as I saw in the document, I could not find a way to do that.
There are some mention of ingestion port but no ingestion URI... 
It would be good to be able to specify 2 different (sets of) URI(s) for each different fonctionality (ingestion & display). Maybe I missed something?

Which brings me to another discussion: is there a redirection mechanism? For example, in the same use case, if some applications are not yet changed to use the https URI, seq answers the call to http and then redirects/forces to https.

KR,
Nicolas

**Comment by nblumhardt** [comment, marked as answer]
Hi Nicolas, thanks for getting in touch!

Assuming your Seq server is `example.com`, the following will:

 * Set up HTTP and HTTPS ingestion ports at `http://example.com:5341` and `https://example.com:45341`, respectively,
 * Host the UI at `http://example.com` and `https://example.com`, and
 * Redirect **non-ingestion** requests for `http://example.com` to `https://example.com` (redirection is not applied to ingestion).

```
seq config set -k api.listenUris -v http://example.com,https://example.com,http://example.com:5341,https://example.com:45341
seq config set -k api.ingestionPorts -v 5341,45341
seq config set -k api.redirectHttpToHttps -v True
seq service restart
```

Let me know if this helps!

Best regards,
Nick

**Reply by ntenoutasse** [reply]
Hi and thank you very much for your answer. It already helps quite a bit!
Just to make sure I understand, this clearly answers my second point, the redirection, which for some reason I didn't see what was in front of me...

And this kind of implies, you cannot have the solution I was trying to achieve in my first point which was, continuing from your assumption:
- host the UI on https://example.com only
- ingest from both http://example.com and https://example.com

Am I correct?

Once again, thanks for you help,
Nicolas

**Reply by nblumhardt** [reply]
Hi Nicolas; I believe you're incorrect - the situation you're describing should work with the above (you can even skip the dedicated ingestion port configuration and additional ingestion URLs, if you don't need them).

Hope this helps,
Nick
